modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/function/ToHexFunction.java

```diff
@@ -0,0 +1,54 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2017, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.mbstyle.function;
+
+import org.geotools.filter.FunctionExpressionImpl;
+import org.geotools.filter.capability.FunctionNameImpl;
+import org.opengis.filter.capability.FunctionName;
+
+import static org.geotools.filter.capability.FunctionNameImpl.parameter;
+
+/**
+ * Generate a hexidecimal value from an rgb value in the range of 0-255.
+ */
+public class ToHexFunction extends FunctionExpressionImpl {
+
+    public static FunctionName NAME = new FunctionNameImpl("tohex",
+            parameter("int", Integer.class));
+
+    public ToHexFunction() {
+        super(NAME);
+    }
+
+    public Object evaluate(Object feature) {
+        Integer arg0;
+
+        try { // attempt to get value and perform conversion
+            Number number = getExpression(0).evaluate(feature, Integer.class );
+            arg0 = number.intValue();
+        } catch (Exception e) {
+            // probably a type error
+            throw new IllegalArgumentException(
+                    "Filter Function problem for function tohex argument #0 - expected type integer");
+        }
+        String hex = Integer.toHexString(arg0);
+        if (hex.length() == 1) {
+            hex = 0 + hex;
+        }
```

Comment:
This implementation raises some issues if we ever want to port it to FilterFunction for more general use, since the 0-padding only makes sense in the context of RGB. If someone wanted just a regular hex conversion this would not quite be useable to to the padding.

Given this, it probably just makes more sense to make a `"torgb"` function instead of `"tohex"`. This would make the generated SLD clearer and more concise as well.